src/Sentry.Hangfire/SentryJobFilter.cs

```diff
@@ -0,0 +1,87 @@
+using Hangfire.Server;
+using Sentry.Extensibility;
+
+namespace Sentry.Hangfire;
+
+internal class SentryJobFilter : IServerFilter
+{
+    internal const string SentryMonitorSlugKey = "SentryMonitorSlug";
+    internal const string SentryCheckInIdKey = "SentryCheckInIdKey";
+
+    private readonly IHub _hub;
+    private readonly IDiagnosticLogger? _logger;
+
+    public SentryJobFilter() : this(() => HubAdapter.Instance)
+    { }
+
+    internal SentryJobFilter(Func<IHub> hubAccessor)
+    {
+        _hub = hubAccessor.Invoke();
```

Comment:
why do we need the accessor?

Originally it was added because the hub could change (e.g: aspnetcore double init). so we called on the accessor on each use to make sure we were on the latest Hub instance

this patterns seems to defeat that purpose